src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationBufferTaggerProvider.Tagger.cs

```diff
@@ -122,9 +122,19 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetAllTags(NormalizedSnapshotSp
                 var snapshot = firstSpan.Snapshot;
                 Debug.Assert(snapshot.TextBuffer == _subjectBuffer);
 
-                var cachedSnapshot = this.CachedSnapshot;
+                // We want to classify from the start of the first requested span to the end of the 
+                // last requested span.
+                var spanToTag = new SnapshotSpan(snapshot,
+                    Span.FromBounds(spans.First().Start, spans.Last().End));
```

Comment:
Likely not an issue for copy, but I could imagine a scenario where you have large collapsed regions between spans  and we end up still classifying a lot.

This is still better than the current code though.